config/initializers/comfy_patching.rb

```diff
@@ -29,7 +29,10 @@ def assign_layout_categories
         end
       end
 
-      delete_orphan_categories && return if _categories.blank?
+      layouts_categories = Comfy::Cms::LayoutsCategory.where(layout_id: self.id)
+      orphan_categories = layouts_categories.to_a - _categories
```

Comment:
@stanleypliu Not sure what this is supposed to do?
`layouts_categories.to_a` is an array containing `LayoutsCategory` objects, while `_categories` is an array of hashes with information about the layout and its tags. So theoretically this operation always returns `layouts_categories.to_a` regardless of what's inside `_categories`

Reply:
You're right - I mistook the `_categories` variable to also be an array of LayoutsCategory objects which were present before the save, didn't pick up on this because it correctly deletes the categories if they aren't present.

Reply:
so i've dispensed with subtracting `_categories` from `layouts_categories` and just settled for filtering `layouts_categories` to get the orphans. one thing i'm not too sure about is how you register new categories tags, it doesn't seem to be as simple as adding the new label after the tag in the CMS editor. If we know we are just going to have topics and types as the two main category types, should we have error handling in case anyone editing the CMS in the future tries to add a new custom label?